src/pages/base.js

```diff
@@ -16,8 +16,18 @@ class Page {
   visitWithParameters(data) {
     let url = this.url;
 
+    const additionalParams = [];
     for (const item of data.raw()) {
-      url = url.replace(`:${item[0]}`, item[1]);
+      if (url.indexOf(`:${item[0]}`) === -1) {
```

Comment:
Maybe it would be good to use reduce instead of this loop, so you can pass this.url as an initial value.